libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -671,8 +666,13 @@ private val fakeDirectionsRoute: DirectionsRoute by lazy {
         .build()
 }
 
-private fun DataRef.asReader(): Reader =
-    InputStreamReader(ByteBufferBackedInputStream(buffer))
+private fun DataRef.toDirectionsResponse(): DirectionsResponse {
+    val stream = ByteBufferBackedInputStream(buffer)
+    val reader = InputStreamReader(stream)
+    val response = DirectionsResponse.fromJson(reader)
+    reader.close()
```

Comment:
Not necessary for the fix, but it felt like it was missing.

Reply:
maybe call close from finally block?